### PR TITLE
{bio}[foss/2017b] Bsoft v2.0.2

### PR DESCRIPTION
**Pull request by JackPerdue**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@JackPerdue We already have easyconfigs for this software under the name `bsoft`, cfr. https://github.com/easybuilders/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/b/bsoft/bsoft-1.8.8-goolf-1.5.14.eb

So we should either rename that to `Bsoft`, or stick with `bsoft`.

The logo suggests `Bsoft`, so I guess the former is the right way to go?

**Comment by JackPerdue** [comment]
I pondered that (and originally built bsoft) but I think "Bsoft" is the correct form given the web site.

**Comment by JackPerdue** [comment]
Note 2.0.2 is hot off the presses.... I've been working with "B"@NIH to come up with a .tgz that will pass Travis's muster... we'll see (the download was updated yesterday).

File: easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    ('libxml2', '2.9.8'),
+    ('Tk', '8.6.8')
+]
+
+skipsteps = ['configure']
+
+prebuildopts = """
+    # fix libxml2 include path
+    sed -e 's,^XMLINC=.*,XMLINC=$EBROOTLIBXML2/include/libxml2,'\
+      -i.eb.xmlinc bsoft/bsoft_conf
```
Comment (boegel) [inline easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb]: @JackPerdue This is a bit... funky (also because of the formatting).

Wouldn't a patch file be better here?

File: easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    # fix fftw3 include path
+    sed -e 's,LIBFFTW/api,LIBFFTW/include,'\
+      -i.eb.fftw3inc bsoft/bsoft_conf
+    # configure
+    ./bconf --prefix=%(installdir)s --fftw=$EBROOTFFTW --tiff=$EBROOTLIBTIFF\
+            --jpeg=$EBROOTLIBJPEGMINTURBO --png=$EBROOTLIBPNG
+"""
+
+# try parallel (EB default) build first then fall back to sequential
+buildopts = 'bsoft || make bsoft'
```
Comment (boegel) [inline easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb]: Why is this necessary?

File: easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+toolchain = {'name': 'foss', 'version': '2017b'}
+toolchainopts = {'openmp': True}
+
+# https://lsbr.niams.nih.gov/bsoft/bsoft2_0_2.tgz
+source_urls = ['http://lsbr.niams.nih.gov/bsoft/']
+sources = ['%%(namelower)s%s.tgz' % '_'.join(version.split('.'))]
+
+dependencies = [
+    # ('FFTW', '3.3.6'),  # FFTW is included in foss/2017b toolchain
```
Comment (boegel) [inline easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb]: OK to remove?

File: easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    ./bconf --prefix=%(installdir)s --fftw=$EBROOTFFTW --tiff=$EBROOTLIBTIFF\
+            --jpeg=$EBROOTLIBJPEGMINTURBO --png=$EBROOTLIBPNG
+"""
+
+# try parallel (EB default) build first then fall back to sequential
+buildopts = 'bsoft || make bsoft'
+
+sanity_check_paths = {
+    'files': ['bsoft.bashrc', 'bsetup', 'bin/brun', 'bin/bshow',
+              'lib/libbsoft.so', 'lib/libbshow.so'],
```
Comment (boegel) [inline easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb]: Please use `SHLIB_EXT` rather than hardcoding `.so`

**Comment by lexming** [comment]
@JackPerdue as part of the merge sprint of old PRs, I have reviewed and updated this PR on Bsoft. I have changed it quite a bit to make it work, but now it is in a good state to be merged:
* fixed all hardcoded paths in configuration files with a patch, there were too many to be done with `sed` commands
* there is a very special case with LibTIFF: Bsoft needs some headers that can only be found in the build directory of LibTIFF. Those headers are specifically flagged as `non-install` in LibTIFF though, so they are never installed. Therefore, I had to make a bundle including a build of LibTIFF to have access to those headers. However, I hacked a little bit that component to not install anything, because those are not needed at runtime. LibTIFF libraries are still taken from the regular LibTIFF dependency.
* LibTIFF has to be build with PIC. For the moment I'm enabling PIC in those LibTIFF easyconfigs belonging to this toolchain or its sub-toolchains. @boegel should I make a PR enabling PIC in all LibTIFF easyconfigs? That's an inoffensive change.
* all other dependencies are taken from EB
* moved execution of `bconf` to the configuration step
* fixed `buildopts` and added `parallel=1`, otherwise build fails due to missing objects
* added `postinstallcmds` to emulate what is done by `Scripts/postinstall`
* enhanced the sanity checks

File: easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb]: fixed in https://github.com/easybuilders/easybuild-easyconfigs/pull/6551/commits/2f0b6aec082a1804e4b094f3537ac3a2cf59cd97

File: easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb (hunk 2 above)
Comment (lexming) [reply easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb]: fixed in https://github.com/easybuilders/easybuild-easyconfigs/pull/6551/commits/2f0b6aec082a1804e4b094f3537ac3a2cf59cd97 by building `bsoft` with `parallel=1`

File: easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb (hunk 3 above)
Comment (lexming) [reply easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb]: done in https://github.com/easybuilders/easybuild-easyconfigs/pull/6551/commits/2f0b6aec082a1804e4b094f3537ac3a2cf59cd97

File: easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb (hunk 4 above)
Comment (lexming) [reply easybuild/easyconfigs/b/Bsoft/Bsoft-2.0.2-foss-2017b.eb]: done in https://github.com/easybuilders/easybuild-easyconfigs/pull/6551/commits/2f0b6aec082a1804e4b094f3537ac3a2cf59cd97

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node101.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/f68790061ad7578038c38dee5471f8d1 for a full test report.

**Comment by JackPerdue** [comment]
@lexming, have at it. :)  Sprint baby sprint!!! I won't get in your way.

**Comment by boegel** [comment]
> @boegel should I make a PR enabling PIC in all LibTIFF easyconfigs?

@lexming makes sense

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node2603.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/c8c0280b5437ca6ddbd6b7a7b50306af for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @JackPerdue!

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 17 out of 25 (18 easyconfigs in total)
node379.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/33998a892b8211cc1f27fba1225abade for a full test report.
